src/components/MessageList/TypingIndicator.js

```diff
@@ -24,74 +22,72 @@ const Container = styled.View`
+  const nonSelfUsers = [];
+  typingKeys.forEach((item, i) => {
+    if (client.user.id === typingUsers[typingKeys[i]].user.id) {
+      return;
+    }
+    nonSelfUsers.push(
+      typingUsers[typingKeys[i]].user.name ||
+        typingUsers[typingKeys[i]].user.id,
+    );
+  });
 
-  constructTypingString = (dict) => {
-    const { client, t } = this.props;
-    const arr2 = Object.keys(dict);
-    const arr3 = [];
-    arr2.forEach((item, i) => {
-      if (client.user.id === dict[arr2[i]].user.id) {
-        return;
-      }
-      arr3.push(dict[arr2[i]].user.name || dict[arr2[i]].user.id);
+  let outStr = '';
+  if (nonSelfUsers.length === 1) {
+    outStr = t('{{ user }} is typing...', { user: nonSelfUsers[0] });
+  } else if (nonSelfUsers.length === 2) {
+    //joins all with "and" but =no commas
+    //example: "bob and sam"
+    outStr = t('{{ firstUser }} and {{ secondUser }} are typing...', {
+      firstUser: nonSelfUsers[0],
+      secondUser: nonSelfUsers[1],
+    });
+  } else if (nonSelfUsers.length > 2) {
+    //joins all with commas, but last one gets ", and" (oxford comma!)
+    //example: "bob, joe, and sam"
+    outStr = t('{{ commaSeparatedUsers }} and {{ lastUser }} are typing...', {
+      commaSeparatedUsers: nonSelfUsers.slice(0, -1).join(', '),
+      lastUser: nonSelfUsers.slice(-1),
     });
+  }
 
-    let outStr = '';
-    if (arr3.length === 1) {
-      outStr = t('{{ user }} is typing...', { user: arr3[0] });
-      dict;
-    } else if (arr3.length === 2) {
-      //joins all with "and" but =no commas
-      //example: "bob and sam"
-      outStr = t('{{ firstUser }} and {{ secondUser }} are typing...', {
-        firstUser: arr3[0],
-        secondUser: arr3[1],
-      });
-    } else if (arr3.length > 2) {
-      //joins all with commas, but last one gets ", and" (oxford comma!)
-      //example: "bob, joe, and sam"
-      outStr = t('{{ commaSeparatedUsers }} and {{ lastUser }} are typing...', {
-        commaSeparatedUsers: arr3.slice(0, -1).join(', '),
-        lastUser: arr3.slice(-1),
-      });
-    }
+  return outStr;
+};
 
-    return outStr;
-  };
+const TypingIndicator = (props) => {
+  const { Avatar = DefaultAvatar } = props;
+  const { client } = useContext(ChatContext);
+  const { typing } = useContext(ChannelContext);
+  const { t } = useContext(TranslationContext);
+  const typingUsers = Object.values(typing);
 
-  render() {
-    const { typing, Avatar, client } = this.props;
-    const typingUsers = Object.values(typing);
+  return (
+    <Container testID='typing-indicator'>
+      {typingUsers
+        .filter(({ user }) => user.id !== client.user.id)
+        .map(({ user }, idx) => (
+          <Avatar
+            image={user.image}
+            key={user.id + idx}
+            name={user.name || user.id}
+            size={24}
+            testID={`typing-avatar-${idx}`}
+          />
+        ))}
+      <TypingText>{constructTypingString(typing, client, t)}</TypingText>
```

Comment:
naming is a bit odd cause we're receiving `typingUsers` in the `contructTypingString` method but we have that variable declared here and aren't passing it in as that
```suggestion
      <TypingText>{constructTypingString({ client, t, typingUsers: typing })}</TypingText>
```